service/db.go

```diff
@@ -462,58 +462,67 @@ func updateUserRow(uuid string, svcDerived *pblib.User, dbDerived *pblib.User) (
 	return updatedUser, nil
 }
 
-//TODO work on later
-func insertNewSecret(secret *pblib.Secret) error {
-	if secret == nil {
-		return authconst.ErrNilSecret
+// insertNewSecret inserts a newly generated secret key to database
+// secret key is used to sign JWT's
+// Returns err if secret is empty or error with database
+func insertNewSecret() error {
+	// generate a new secret
+	secretKey, err := generateToken(auth.SecretByteSize)
+	if err != nil {
+		return err
 	}
 
 	command := `
 				INSERT INTO user_security.secret(
-					secret_key, created_timestamp
-				) VALUES($1, $2)
+					secret_key, created_timestamp, expiration_timestamp
+				) VALUES($1, $2, $3)
 				`
 
-	_, err := postgresDB.Exec(command, secret.Key, secret.CreatedTimestamp)
-
+	expirationTimestamp, err := generateSecretExpirationDate(time.Now().UTC())
 	if err != nil {
 		return err
 	}
 
-	return nil
-}
-
-//TODO work on later
-func insertJWToken(uuid string, header *auth.Header, body *auth.Body, token string, secretKey string) error {
-	if err := validateUUID(uuid); err != nil {
-		return consts.ErrInvalidUUID
-	}
-	if header == nil {
-		return authconst.ErrNilHeader
-	}
-	if body == nil {
-		return authconst.ErrNilBody
-	}
-	if token == "" {
-		return authconst.ErrEmptyToken
-	}
-	if secretKey == "" {
-		return authconst.ErrEmptySecret
-	}
-
-	command := `
-				INSERT INTO user_security.tokens(
-					token_string, secret_key, token_type, algorithm,
-					permission, expiration_date, uuid
-				) VALUES($1, $2, $3, $4, $5, $6, $7)
-				`
-
-	_, err := postgresDB.Exec(command, token, secretKey, header.TokenTyp, header.Alg,
-		auth.PermissionStringMap[body.Permission], time.Unix(body.ExpirationTimestamp, 0), uuid)
+	_, err = postgresDB.Exec(command, secretKey, time.Now().UTC(), expirationTimestamp)
 
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
+
+//TODO work on later
```

Comment:
A change like this need to be force committed to a master to lessen the noise in code review.

Reply:
i will do that next time